### ISSUE TITLE
"Repository successfully updated" even though it wasn't

### ISSUE THREAD

**Issue by iliis** [KSP-CKAN/CKAN#1389]
https://codeload.github.com/KSP-CKAN/CKAN-meta/tar.gz/master seems to be currently offline (500: Internal Server Error) at the time of writing. This causes the mod list to be completely empty without any indication of what went wrong. The log in the "updating repositories tab" (and the status bar at the bottom) even shows success:

```
Connection to https://github.com/KSP-CKAN/CKAN-meta/archive/master.tar.gz could not be established.
Repository successfully updated
```
### Steps to reproduce
- take GitHub offline (well, block the metadata repository somehow ;))
- click "refresh"
### Expected outcome

Some sort of error message. At least something like "Sorry, couldn't update the metadata repository." Preferably showing the old list of mods without clearing the list.
### Actual outcome

Empty list of mods (even if it was successfully updated beforehand), no immediate indication of what went wrong.
### Version

CKAN v1.12.0-g0862b9b (current stable version, even tough it's marked as beta in About)
KSP 1.0.4 (from Steam)
OS: Windows 7

**Comment by cgruber** [comment]
Any progress here?  I'm not even getting hte "connection to" message - it just hangs.

**Comment by HebaruSan** [comment]
This was fixed in #2277; we now report failure if it fails.
